server/controllers/finance/journal/index.js

```diff
@@ -13,10 +13,71 @@
 
 // module dependencies
 const db = require('../../../lib/db');
+const core = require('./core');
+
+var uuid = require('node-uuid');
 
 // expose to the api
 exports.list = list;
 
+exports.reverse = reverse;
+
+
+// GET /Grade
+function lookReverseTransaction(uid, user_id) {
+  'use strict';
+
+  let transaction = db.transaction();
+
+
+  let sql = `
+    INSERT INTO posting_journal (uuid, project_id, fiscal_year_id, period_id,
+      trans_id, trans_date, record_uuid, description, account_id, debit,
+      credit, debit_equiv, credit_equiv, currency_id, entity_uuid,
+      entity_type, reference_uuid, comment, origin_id, user_id, cc_id, pc_id)
+    SELECT
+      HUID(UUID()), p.project_id, @fiscalId, @periodId, @transId, p.trans_date,
+      p.record_uuid, 'Credit Note', p.account_id, p.debit * -1, p.credit * -1,
+      p.debit * @exchange * -1, p.credit * @exchange * -1, p.currency_id,
+      p.entity_uuid, 'C', p.reference_uuid, NULL, 1, p.user_id, NULL, NULL
+    FROM posting_journal AS p
+    WHERE p.record_uuid = ?;`;
+
+  transaction
+
+    // set up the SQL variables for core.js to consume
+    .addQuery(`
+      SELECT posting_journal.trans_date, enterprise.id, project.id, posting_journal.currency_id
+      INTO @date, @enterpriseId, @projectId, @currencyId
+      FROM posting_journal JOIN project JOIN enterprise ON
+        posting_journal.project_id = project.id AND
+        project.enterprise_id = enterprise.id
+      WHERE posting_journal.record_uuid = ? LIMIT 1;
+    `, [uid])
+
+    .addQuery(`
+      UPDATE invoice SET is_credit_note = 1, credit_note_by = ${user_id} 
+      WHERE uuid = ?
+    `, [uid]);
+
+
+  // this function sets up the dates, fiscal year, and exchange rate for this
+  // posting session and ensures they all exist before attempting to write to
+  // the posting journal.
+  core.setup(transaction);
+
+  transaction
+    .addQuery(
+      sql,[uid]
+    );
+
+  return transaction.execute()
+    .then(function (rows) {
```

Comment:
This `.then()` function doesn't do anything.  Please remove it.

@IMA-WorldHealth/local-contributors, this is a pattern I see a lot in code being submitted in PRs.  The following are equivalent:
```js
return transaction.execute();
// ===
return transaction.execute()
  .then(function (rows) {
    return rows;
  });
```

In fact, you could do this over and over.
```js
return $http.get('/some/route')
  .then(function (a) { return a; })
  .then(function (b) { return b; })
  .then((c) => { return c; })
  .then(d => d)
  .then((e) => e)
  .then(function (f) { return f; })
  // .. etc ...
  .then(function (z) { return z; });
```
All of the functions of the form `.then(function (x) { return x; });` do not add any value to the code and should be removed.